### ISSUE TITLE
Make it possible to use different types of pluggable transports

### ISSUE THREAD

**Issue by A-Shahbazi** [Gedsh/InviZible#233]
Currently if we switch between different pluggable transports (e.g. obfs3 to snowflake), only the enabled bridges of currently shown pluggable transport are used in tor configuration.
I think it should be possible to select multiple bridges of multiple types via GUI.

**Comment by A-Shahbazi** [comment]
> > I think it should be possible to select multiple bridges of multiple types via GUI.
> 
> It is impossible. Tor configuration does not allow this. Only one pluggable transport can be used at a time.

This is not correct. The point is the torrc config file needs to have relevant config line for each pluggable transport and then bridges of different pluggable transports can be used simultaneously. For example the following is a working torrc config:
```
...
ClientTransportPlugin obfs3,obfs4,scramblesuit,meek_lite,webtunnel exec /usr/bin/lyrebird-pt-client

ClientTransportPlugin snowflake exec /usr/bin/snowflake-pt-client -max 10 -log snowflake.log -ice stun:stun.l.google.com:19302,stun:stun.antisip.com:3478,stun:stun.bluesip.net:3478,stun:stun.dus.net:3478,stun:stun.epygi.com:3478,stun:stun.sonetel.com:3478,stun:stun.sonetel.net:3478,stun:stun.stunprotocol.org:3478,stun:stun.uls.co.za:3478,stun:stun.voipgate.com:3478,stun:stun.voys.nl:3478 -log-to-state-dir -unsafe-logging

Bridge snowflake 192.0.2.3:80 2B280B23E1107BB62ABFC40DDCC8824814F80A72 fingerprint=2B280B23E1107BB62ABFC40DDCC8824814F80A72 utls-imitate=hellorandomizedalpn

Bridge obfs3 109.105.109.163:38980 1E05F577A0EC0213F971D81BF4D86A9E4E8229ED

Bridge obfs4 85.31.186.98:443 011F2599C0E9B27EE74B353155E244813763C3E5 cert=ayq0XzCwhpdysn5o0EyDUbmSOx3X/oTEbzDMvczHOdBJKlvIdHHLJGkZARtT4dcBFArPPg iat-mode=0
```

**Comment by A-Shahbazi** [comment]
> > > I think it should be possible to select multiple bridges of multiple types via GUI.
> > 
> > 
> > It is impossible. Tor configuration does not allow this. Only one pluggable transport can be used at a time.
> 
> This is not correct. The point is the torrc config file needs to have relevant config line for each pluggable transport and then bridges of different pluggable transports can be used simultaneously. For example the following is a working torrc config:
> 
> ```
> ...
> ClientTransportPlugin obfs3,obfs4,scramblesuit,meek_lite,webtunnel exec /usr/bin/lyrebird-pt-client
> 
> ClientTransportPlugin snowflake exec /usr/bin/snowflake-pt-client -max 10 -log snowflake.log -ice stun:stun.l.google.com:19302,stun:stun.antisip.com:3478,stun:stun.bluesip.net:3478,stun:stun.dus.net:3478,stun:stun.epygi.com:3478,stun:stun.sonetel.com:3478,stun:stun.sonetel.net:3478,stun:stun.stunprotocol.org:3478,stun:stun.uls.co.za:3478,stun:stun.voipgate.com:3478,stun:stun.voys.nl:3478 -log-to-state-dir -unsafe-logging
> 
> Bridge snowflake 192.0.2.3:80 2B280B23E1107BB62ABFC40DDCC8824814F80A72 fingerprint=2B280B23E1107BB62ABFC40DDCC8824814F80A72 utls-imitate=hellorandomizedalpn
> 
> Bridge obfs3 109.105.109.163:38980 1E05F577A0EC0213F971D81BF4D86A9E4E8229ED
> 
> Bridge obfs4 85.31.186.98:443 011F2599C0E9B27EE74B353155E244813763C3E5 cert=ayq0XzCwhpdysn5o0EyDUbmSOx3X/oTEbzDMvczHOdBJKlvIdHHLJGkZARtT4dcBFArPPg iat-mode=0
> ```

I tested using `Edit tor.conf directly` on app settings and I was able to set one webtunnel bridge and another snowflake bridge simultaneously and it's working.

**Comment by A-Shahbazi** [comment]
@Gedsh, can you take another look into this?

**Comment by Gedsh** [comment]
I configured Tor to use obfs4 and webtunnel bridges at the same time. It doesn't work. Tor can't connect. I performed this test just now.

**Comment by A-Shahbazi** [comment]
> I configured Tor to use obfs4 and webtunnel bridges at the same time. It doesn't work. Tor can't connect. I performed this test just now.

What was the error message? Did you add both "obfs4" and "webtunnel" to the line defining ClientTransportPlugin(like what I did [here](https://github.com/Gedsh/InviZible/issues/233#issuecomment-2304234531))? Can you provide relevant torrc lines?

**Comment by A-Shahbazi** [comment]
I tested again and here's the working torrc config I tested using 2.3.2-beta version:
```
...
UseBridges 1

ClientTransportPlugin obfs4 exec /data/app/~~XIcL2bUloqK2zpExOAJ1Wg==/pan.alexander.tordnscrypt-2rQEXhUBKRLFAPOzwptFpw==/lib/arm64/libobfs4proxy.so

ClientTransportPlugin webtunnel exec /data/app/~~XIcL2bUloqK2zpExOAJ1Wg==/pan.alexander.tordnscrypt-2rQEXhUBKRLFAPOzwptFpw==/lib/arm64/libwebtunnel.so -log /data/user/0/pan.alexander.tordnscrypt/logs/WebTunnel.log

Bridge webtunnel [2001:db8:c744:83a9:b870:eb3a:947b:9be8]:443 D8B44D31E5719552DB634F58A0A4C96187FA5EF4 url=https://rusthelp.org/och8theishaephisheeX

Bridge obfs4 109.105.109.147:13764 BBB28DF0F201E706BE564EFE690FE9577DD8386D cert=KfMQN/tNMFdda61hMgpiMI7pbwU1T+wxjTulYnfw+4sgvG0zSH7N7fwT10BI8MUdAD7iJA iat-mode=2```
